client/my-sites/plugins/controller.js

```diff
@@ -74,7 +75,8 @@ function renderSinglePlugin( context, siteUrl, isWpcomPlugin ) {
 			isWpcomPlugin: isWpcomPlugin,
 			onPluginRefresh: title => titleActions.setTitle( title )
 		} ),
-		document.getElementById( 'primary' )
+		document.getElementById( 'primary' ),
+		context.store
```

Comment:
> we build some widget that only depends on a few props but instead of making it just that we end up connecting that component to the store.

Well, my rule of thumb would be, that if we need to create some data model for anything and it needs to be accessible from different parts of render tree, it needs to end up somewhere.
For now we are just moving content from flux stores to redux stores and we are not connecting 'too much' I hope.
I agree with you and would be very sceptical about storing for example all of internal components state in redux store and I heard some chatter on the web about it.
So, if component is performing one function and logically is a part of a bigger picture, I'm all for passing the data via props.
But if we need to create a state component which is mostly for storing some data and has no logical function other then that, then I believe this data should end up in redux.

Like everything in life, this requires balance and common sense :)

>it gets called by the store on every update and we get those crazy re-render chains again 

Well, this actually is not true, redux is keeping track of all / new props for the exact subtree you selected in connect:
https://github.com/rackt/react-redux/blob/master/src/components/connect.js#L101
The good way to think about it is that you kinda get `PureRenderMixin` for free.
So, only the compoennts whose redux store subtree was changed, get re-rendered.

Also, I feel like we have moved this discussion a bit from the original point.
I really dont see any other place for the store to be, then current page refresh context.
I am happy to discuss it, but I dont see any other option (then singleton which was discarded by theme showcase folks)